lib/stylesheet/compiler.rb

```diff
@@ -65,14 +65,25 @@ def self.compile(stylesheet, filename, options = {})
       result = engine.render
 
       if options[:rtl]
-        require "r2"
-        [R2.r2(result), nil]
+        [rtlify_css(result) || result, nil]
       else
         source_map = engine.source_map
         source_map.force_encoding("UTF-8")
 
         [result, source_map]
       end
     end
+
+    def self.rtlify_css(src_css)
+      @context ||=
+        begin
+          context = MiniRacer::Context.new
+          context.eval(
+            File.read("#{Rails.root}/app/assets/javascripts/rtlcss-miniracer/dist/main.js"),
+          )
+          context
+        end
+      @context.eval("rtlcss.process(#{src_css.inspect})")
```

Comment:
`.inspect` does not always make an object safe to interpolate into JS. `.to_json` does.
```suggestion
      @context.eval("rtlcss.process(#{src_css.to_json})")
```

Reply:
Or even better - use `@context.call` to avoid the need for interpolation and improve performance: https://github.com/rubyjs/mini_racer#function-call

(In core, we do this a lot in [discourse_js_processor.rb](https://github.com/discourse/discourse/blob/b96869d5fb7702f74b8191d3fb0108d68ff0df51/lib/discourse_js_processor.rb#L252-L252))